src/passes/shadowstack.ts

```diff
@@ -658,6 +658,7 @@ class InstrumentReturns extends Pass {
     var stmts = new Array<ExpressionRef>();
     if (value) {
       let returnType = _BinaryenExpressionGetType(value);
+      assert(returnType != TypeRef.Unreachable, "shadowstack pass cannot handle return unreachable");
```

Comment:
If the returned value is unreachable, code generated here cannot be reached anyhow it seems:
```ts
let t = unreachableExpr; // aborts execution
__stack_pointer += frameSize
return t
```
As such, wdyt of covering this issue in the shadow pass directly, say by, if `returnType` is unreachable, doing nothing:
```ts
// Nothing to do if execution terminates before a value is returned
if (returnType == TypeRef.Unreachable) return;
```
Then other code can remain as-is, relying on the assumption that an unreachable value can substitute a concrete value.

Reply:
The new commit has fixed it 😄

Reply:
With the check now in the shadow stack pass, can the other changes be reverted? :)

Reply:
Yes, testcase is passing after trying to revert other change. 😃